### ISSUE TITLE
rules error: Mons Insurance should lose 3 MC for any "actual" attack on the neutral opponent

### ISSUE THREAD

**Issue by kevinb9n** [terraforming-mars/terraforming-mars#3342]
<img src=https://user-images.githubusercontent.com/934551/120847402-4a3ff000-c528-11eb-9fc9-e950539b4fb4.png width=250> <img src=https://user-images.githubusercontent.com/934551/120847534-778c9e00-c528-11eb-928a-60bc53c88cce.png width=250>

(I also checked if the same happens when I have 3+ MC, and it does.)

In the solo game, even though attacks against the neutral "player" are invisible, they are always treated as having actually happened, and thus Mons is forced to pay.

In case of a resource attack, the Mons player will want to choose whether to attack or not; it may sometimes be worth the 3 MC (such as with a steal attack), sometimes not.

**Comment by kevinb9n** [comment]
Oh, actually even in this case, I should technically have had the option of attack myself, in case for some reason I would rather lose 2 plant production than any megacredits.

**Comment by kevinb9n** [comment]
More examples

<img width="698" alt="image" src="https://user-images.githubusercontent.com/934551/162596968-511399f2-bb74-46ed-a457-62207bf0df55.png">

If I play these 3 cards (stealing steel) I should be 5 MC poorer from the costs and also 9 MC poorer from the Mons effect. But...

<img width="698" alt="image" src="https://user-images.githubusercontent.com/934551/162596939-4b9b765f-83ce-4576-b26b-8bb866d6c9e3.png">

Basically, if I choose Mons in solo, the app sorta forces me to cheat, in a way.

the receipts: https://boardgamegeek.com/thread/2187014/article/32049249#32049249

**Comment by kberg** [comment]
Thanks for the report. I think this is solveable. I see the pieces where it needs to be fixed.

**Comment by kberg** [comment]
This would be much easier if the neutral player was an actual player.

**Comment by kevinb9n** [comment]
Finally had my first game with Mons.

I think that if I play a card like Asteroid which has an *optional, non-stealing* attack (i.e., all "remove any player's resource" cards) then I would *virtually always* just choose not to attack to save the 3 MC. (There could be a rare exception but screw it.) What the app was doing was fine in *these* cases.

This issue was trying to cover the other 2 cases:

For a *stealing* attack I would always expect the stealing to happen (I won't come out behind on net balance, and would just not play the card if so, anyway). On Air Raid I come out 2 MC ahead, for example. It's probably working this way now, I guess.

For a "decrease any player's production", well, the unfortunate fact is the player needs to choose, because sometimes they'd rather lose the 3 MC and sometimes they'd rather lose their own production. But, I do think losing the MC is a reasonable guess for what they player wants more often, if it must guess.

Sorry this is so fiddly.

**Comment by kevinb9n** [comment]
Gonna reopen this for now but LMK what I should do.

**Comment by kevinb9n** [comment]
Should this be reopened?

**Comment by kevinb9n** [comment]
To be more plain: 

Simplest fix: when a Solo Mons Insurance player plays an ordinary "remove any player's resources" card (like Asteroid), they should not be charged 3 MC, as in real life they would simply choose not to do the removal.

Probably Proper fix: Really, Mons Insurance player should always have the option of whether to attack the solo player or not (well, except for a decrease-any-production that the player can't fulfill themselves, I guess).

That would address a whole bunch of edge cases involving Ants or Predators or Crash Site Cleanup etc.

**Comment by kberg** [comment]
It will take a while to front-load how this works again, after which I'll read this more carefully. You can help by giving me some details of your game experience e.g. "This happened and is right" "This happened and is wrong" "This happened and is better" "This happened and is worse"

**Comment by kevinb9n** [comment]
I played Mons Insurance. I played Comet. I was charged 3 MC.

<img width="350" alt="Screen Shot 2022-07-30 at 9 13 39 AM" src="https://user-images.githubusercontent.com/934551/181925851-950ef523-c413-4963-853c-53c44971d16a.png">

This is wrong because removing plants with Comet is *optional* (and there is *almost* no reason I would ever want to carry it out).

The *correct* fix would be to let a solo player always choose whether to remove resources from the neutral opponent or not, whenever it is optional. (Specifically: when it is resource-not-production removal-or-stealing, from "any player" (red outline), expressed with a minus sign, *not* when it is an action cost like Ants/Predators. Yes, I am also stunned at what a fiddly persnickety rule it is.)

The *better* fix would be to expose that option only when the player has played Mons Insurance, and otherwise always "do" the attack (that part matters only because it might power up Crash Site Cleanup), because there's no other reason (unless it is from the fan expansions!) why you would even be able to tell the difference.

But a probably-simple(?) fix that would make things a *lot* better than now is: if solo player has played Mons Insurance, then assume NO removal whenever it is optional (as above) and is not a "steal".

I hope this helps.

**Comment by kevinb9n** [comment]
FWIW, this seems to be the only bug left that ever really gets in my face, and if nothing else I think rolling back the original fix would be better than how it is now (i.e. rollback would be the 4th best solution by my reckoning).

**Comment by kberg** [comment]
I see what's going on. You're right that it's possible to have cards like comet not remove plants. The only reason a player might still want this is to play Crash Site Cleanup. That is probably manageable, too.

**Comment by kberg** [comment]
There are, it seems, a few other places where Mons Insurance is actually ignored in solo mode. (Virus, for instance.)
